# Assignment 3- Hypothesis_Testing 

## Question-1
A F&B manager wants to determine whether there is any significant difference in the diameter of the cutlet between two units. A randomly selected sample of cutlets was collected from both units and measured? Analyze the data and draw inferences at 5% significance level. Please state the assumptions and tests that you carried out to check validity of the assumptions.

In [1]:
#import the libraries
import pandas as pd
import scipy                        
import numpy as np
from scipy import stats  

First, we will have to define the null and alternate hypothesis:<br>

$H_{null}$ : There is no significant difference in the diameter of the cutlet between two units <br>
$H_{alternate}$ : There is a significant difference in the diameter of the cutlet between two units


Since we are comparing the means of 2 independent groups, we will have to use 2 sample T-test

In [2]:
cutlet_data = pd.read_csv('Cutlets.csv')
cutlet_data.head()

Unit A  Unit B
0  6.8090  6.7703
1  6.4376  7.5093
2  6.9157  6.7300
3  7.3012  6.7878
4  7.4488  7.1522

In [3]:
data1= cutlet_data['Unit A']
data2= cutlet_data['Unit B']

In [4]:
import statsmodels.api as sm
rng = np.random.default_rng()
np.random.Generator

numpy.random._generator.Generator

In [5]:
stats.ttest_ind(data1,data2)

Ttest_indResult(statistic=0.7228688704678061, pvalue=0.4722394724599501)

In [6]:
p=stats.ttest_ind(data1,data2)[1]
p

0.4722394724599501

In [7]:
if p<0.05:
    print("Reject null hypothesis")
else:
    print("Accept null hypothesis")

Accept null hypothesis


Here,pvalue= 0.47223947 is greater than 0.05, so we accept H0 <br>
Hence, we can conclude that there is significant difference in the diameter of the cutlet between two units.